src/openrct2/ride/Ride.cpp

```diff
@@ -1008,18 +1008,13 @@ static void ride_remove_cable_lift(Ride* ride)
     if (ride->lifecycle_flags & RIDE_LIFECYCLE_CABLE_LIFT)
     {
         ride->lifecycle_flags &= ~RIDE_LIFECYCLE_CABLE_LIFT;
-        uint16_t spriteIndex = ride->cable_lift;
-        do
+
+        for (Vehicle* vehicle = GetEntity<Vehicle>(ride->cable_lift); vehicle != nullptr;
+             vehicle = GetEntity<Vehicle>(vehicle->next_vehicle_on_train))
```

Comment:
unfortunately we can't do this due to the `sprite_remove`. Its a bit of an annoying situation that will be fixed but you will need to revert this change for now. Basically any loop that contains a `sprite_remove` has to be handled very carefully as it can invalidate the iterators.